### PR TITLE
Add asm and analysis plugin for kvx architecture

### PR DESCRIPTION
**Pull request by jmaselbas**
KVX is VLIW architecture, kind of like a DSP, it can execute more than
one "instruction" per cycle. Each cycle a bundle of instruction is
executed which is comprised of up to 6 instructions. The general rules
for bundle are: only one BCU instruction (which comes first); only one
LSU instruction; and up to 4 ALU instructions (depending on the type).

In assembly the end of a bundle is marked by a `;;` in an empty line,
here in r2 (to make things more compact and readable) the end of bundle
is appended after the last instruction, but beware that this is not
valid kvx assembly.

---

original PR https://github.com/rizinorg/rizin/pull/2545

### CODE REVIEW

**Comment by jmaselbas** [comment]
hold on before merging this, I've not tried to compiled this... (I need to install rizin first)

**Comment by jmaselbas** [comment]
OK, I've now tested this. There are two plugins added here, one for the disassembly (asm) and the other for the analysis.

**Review by XVilka — CHANGES_REQUESTED** [review]
Sorry for the delay, missed the notification. Overall looks good, but we switched from makefiles to Meson: https://rizin.re/posts/why-meson/

These days Rizin now even can be built with [Muon](https://sr.ht/~lattis/muon/): https://github.com/rizinorg/rizin/blob/dev/.github/workflows/muon.yml#L52

You can see the example how it's done in https://github.com/rizinorg/rizin-extras/blob/master/jaguar-gpu/meson.build

**Comment by jmaselbas** [comment]
Could you provide an example on how meson is used, for example how to build jaguar-gpu plugin.

**Comment by jmaselbas** [comment]
keep in mind that there is an additional target in the makefile to generate opc.kvx from binutils sources, I didn't try to do the same with meson as it is not intended to be ran automatically

File: kvx/meson.build [hunk 1]
```diff
@@ -0,0 +1,20 @@
+project('kvx', 'c')
+
+rz_asm_dep = dependency('rz_asm')
+rz_analysis_dep = dependency('rz_asm')
```
Comment (wargio) [inline kvx/meson.build]: Probably you want to use `rz_analysis_dep = dependency('rz_analysis')` ?

File: kvx/meson.build [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+plugins_dir = get_option('prefix') / rz_asm_dep.get_variable(pkgconfig: 'plugindir', cmake: 'rz_asm_PLUGINDIR')
```
Comment (wargio) [inline kvx/meson.build]: change this to
```py
rizin_plugdir = get_option('rizin_plugdir')
if rizin_plugdir == ''
  plugins_dir = rz_core_dep.get_variable(pkgconfig: 'plugindir', cmake: 'rz_asm_PLUGINDIR')
endif
```

and create a new file like this one https://github.com/rizinorg/rz-libyara/blob/main/meson_options.txt containing just the `rizin_plugdir` line

File: kvx/meson.build (hunk 1 above)
Comment (jmaselbas) [reply kvx/meson.build]: yes, right.

File: kvx/meson.build (hunk 2 above)
Comment (jmaselbas) [reply kvx/meson.build]: ok

**Review by wargio — APPROVED** [review]

File: kvx/meson.build [hunk 3]
```diff
@@ -0,0 +1,23 @@
+project('kvx', 'c')
+
+rz_asm_dep = dependency('rz_asm')
+rz_analysis_dep = dependency('rz_analysis')
+rizin_plugdir = get_option('rizin_plugdir')
+if rizin_plugdir == ''
+  plugins_dir = rz_core_dep.get_variable(pkgconfig: 'plugindir', cmake: 'rz_asm_PLUGINDIR')
```
Comment (wargio) [inline kvx/meson.build]:
```suggestion
  plugins_dir = rz_asm_dep.get_variable(pkgconfig: 'plugindir', cmake: 'rz_asm_PLUGINDIR')
```

**Comment by wargio** [comment]
i have tested the build and does show up
```
$ meson --prefix=~/.local build
The Meson build system
Version: 0.62.0
Source dir: /tmp/rizin-extras/kvx
Build dir: /tmp/rizin-extras/kvx/build
Build type: native build
Project name: kvx
Project version: undefined
C compiler for the host machine: cc (gcc 12.1.0 "cc (GCC) 12.1.0")
C linker for the host machine: cc ld.bfd 2.38
Host machine cpu family: x86_64
Host machine cpu: x86_64
Found pkg-config: /usr/bin/pkg-config (1.8.0)
Run-time dependency rz_asm found: YES 0.5.0
Run-time dependency rz_analysis found: YES 0.5.0
Message: Plugins install directory: lib/rizin/plugins
Build targets in project: 2

kvx undefined

  User defined options
    prefix: ~/.local

Found ninja-1.10.2.git.kitware.jobserver-1 at /usr/bin/ninja
$ ninja -C build/                                                           
ninja: Entering directory `build/'
[8/8] Linking target libkvx_analysis.so
$ ninja -C build/ install
ninja: Entering directory `build/'
[0/1] Installing files.
Installing libkvx_asm.so to /home/deroad/.local/lib/rizin/plugins
Installing libkvx_analysis.so to /home/deroad/.local/lib/rizin/plugins
$ rz-asm -L | grep Kalray
_dA_  32 64      kvx         LGPL3   Kalray VLIW core
```

**Comment by jmaselbas** [comment]
Do you want me to add a version ?

**Comment by wargio** [comment]
> Do you want me to add a version ?

It is up to you, the version number is optional.

**Comment by jmaselbas** [comment]
> It is up to you, the version number is optional..

The actual plugins doesn't have a version number right ? If the version is not exported somehow then, yeah I don't really see to point to have a version number, and this can be merged :)

**Comment by wargio** [comment]
no, i think is good as it is.

**Pull request merged**